### ISSUE TITLE
What is the common way to implement ideas from paper?

### ISSUE THREAD

**Issue by 20092136** [PaddlePaddle/Paddle#3470]
As a beginner of deep learning platform, I have learned how to use paddle python-interface and ready-made component(e.g. layers) to build neural network. The next stage, I am thinking about grasp the skill of implement ideas under paddle's design, for example, modify the Loss Function.

Is there any common way to learn these skill? What is your suggestion?
Is there any prerequisite knowledge I have to learn first such as the paddle design? If so, Is there any materials for us to study?

I'd just like to understand the learning methodology and your suggestion, I'll deal with the details by myself.

**Comment by typhoonzero** [comment]
You can check out this document here: http://doc.paddlepaddle.org/doc/howto/dev/new_layer_en.html
